cloud-services/mbl-cloud-client/scripts/arm_update_activate.sh

```diff
@@ -194,6 +236,23 @@ if echo "${FIRMWARE_FILES}" | grep .ipk$; then
 
     exit "$?"
 
+elif BOOTLOADER2_FILE=$(echo "${FIRMWARE_FILES}" | grep "${WKS_BL2_FILENAME}"); then
+
+    # ------------------------------------------------------------------------------
+    # Write a bootloader 2 component from an update payload file to disk
+    # ------------------------------------------------------------------------------
+    # Find the offset from the 'part-info' file
+    OFFSET=$(cat "${PART_INFO_FILES_DIR}"/MBL_WKS_BOOTLOADER2_OFFSET_BANK1_KiB)
```

Comment:
We should check that `cat` succeeded.

I also suggest that you check whether MBL_WKS_BOOTLOADER2_SKIP exists and contains "1". If it is then we should fail the update.